File: GameEngine_IOS/GameEngine/Models/ComplexEntities/GenericEntity.swift
import Foundation

/**
 * Generic entity without any specific of a determined entity
 */
public class GenericEntity  {
    private static var newId : Int = 0
    
    /**
     * Identifier of the generic entity
     */
    let id : String;
    
    /**
     * Keys: Have the name of the group The name of the material group for
     * instance harm
     */
    let groupsOfMaterials : Dictionary<String, MaterialGroup>;
    
    /**
     * The type of the object that the model is supporting
     */
    let objectType : TEntity;
    
    /**
     * Initiator of the entity to be render in the 3D world
     *
     * @param groupsOfMaterials
     *            HashMap with groups of materials to use in entity
     * @param objectType
     *            The type of the object that the model is supporting
     */
    public init(_ groupsOfMaterials : Dictionary<String, MaterialGroup>, _ objectType : TEntity) {
        self.id = String(GenericEntity.newId);
        GenericEntity.newId += 1;
        self.groupsOfMaterials = groupsOfMaterials;
        self.objectType = objectType;
    }
}

File: GameEngine_PC/GameEngineOpenGL3_2/src/main/java/gameEngine/models/complexEntities/GenericEntity.java
package gameEngine.models.complexEntities;

import java.util.HashMap;

/**
 * Generic entity without any specific of a determined entity
 */
public class GenericEntity {

	/**
	 * Keys: Have the name of the group The name of the material group for
	 * instance harm
	 */
	private final HashMap<String, MaterialGroup> groupsOfMaterials;

	/**
	 * The type of the object that the model is supporting
	 */
	private TEntity objectType;

	/**
	 * 
	 * Constructor of the generic entity to be render in the 3D world
	 * 
	 * @param groupsOfMaterials
	 *            HashMap with groups of materials to use in entity
	 * @param objectType
	 *            The type of the object that the model is supporting
	 */
	public GenericEntity(HashMap<String, MaterialGroup> groupsOfMaterials, TEntity objectType) {
		super();
		this.groupsOfMaterials = groupsOfMaterials;
		this.objectType = objectType;
	}

	/**
	 * @return the groups of materials
	 */
	public HashMap<String, MaterialGroup> getGroupsOfMaterials() {
		return groupsOfMaterials;
	}

	/**
	 * @return The type of the object that the model is supporting
	 */
	public TEntity getObjectType() {
		return objectType;
	}

	
}